cacheops/redis.py

```diff
@@ -13,32 +16,39 @@
 logger = logging.getLogger(__name__)
 
 # Global circuit-breaker flag. Values:
-#   None = circuit breaker closed, Redis calls are attempted.
+#   0 = circuit breaker closed, Redis calls are attempted.
 #   <time_seconds> = circuit breaker open, Redis calls are skipped.
-circuit_breaker_opened = None
+try:
+    fd = os.open('/tmp/redis_circuit_breaker', os.O_RDWR)
+    buf = mmap.mmap(fd, 1024)
+except (OSError, ValueError):
+    logger.warn("Mmap's not working, Redis circuit breaker will be local")
+    circuit_breaker = lambda: None  # noqa: E731
+    circuit_breaker.value = 0
+else:
+    circuit_breaker = ctypes.c_int.from_buffer(buf)
 
 if settings.CACHEOPS_DEGRADE_ON_FAILURE:
     @decorator
     def handle_connection_failure(call):
         """Skip Redis calls for a configurable period after a timeout."""
-        global circuit_breaker_opened
-        if settings.FEATURE_CACHEOPS_CIRCUIT_BREAKER and circuit_breaker_opened:
+        if settings.FEATURE_CACHEOPS_CIRCUIT_BREAKER and circuit_breaker.value:
             # Circuit breaker is open! Should we close it yet?
-            if time() - circuit_breaker_opened > settings.REDIS_CIRCUIT_BREAKER_RESET_SECONDS:
+            if time() - circuit_breaker.value > settings.REDIS_CIRCUIT_BREAKER_RESET_SECONDS:
                 # Yes, let's close the circuit breaker
                 logger.info("Closing Redis circuit breaker")
-                circuit_breaker_opened = None
+                circuit_breaker.value = 0
             else:
-                # No, just skip this Redis call
+                # No, just skip this Redis call and emulate a cache miss
                 logger.debug("Redis circuit breaker is open! Skipping Redis call")
```

Comment:
That means that the *default* log level is `INFO`, so yes unless it's overloaded in the env that debug message won't be logged, which is what we want.